### ISSUE TITLE
'Line Length Violation' VS Xcode's indentation

### ISSUE THREAD

**Issue by DaniSpringer** [realm/SwiftLint#2525]
- [Updated SwiftLint version] how do I check what the latest version is?
- Searched for existing issues: yes.

Xcode usually handles indentation, for example if one adds a newline in between the properties of a UIButton, like so:

```
let myButton = UIButton(title: "Hello", color: purple)
```

Xcode will automatically do this:

```
let myButton = UIButton(title: "Hello",
                        color: purple)
```

But with class definitions, like `class: UIViewController, blah, blah, andBlah`, adding a newline in between properties doesn't keep indentation at all. In fact, adding a newline in between the first and second property will add some indentation (although not the correct amount), while adding another newline in between the second and third property (and the same for the rest) won't add indentation at all.

Based on this, it is my understanding that Xcode devs don't believe class definitions should have Line Length limits.

For me to get a warning when indentation is not supported unfortunately isn't helpful.

**Comment by jpsim** [comment]
I'm not sure I'm following you. You can totally indent your class declarations mid-way through your list of conformances:

```swift
class MyClass: UIViewController, blah, blah
    blah, blah, blah {
    // MyClass members...
}
```

Are you talking about something else?

**Comment by igroti** [comment]
Unfortunately I’m not fluent with the terms for the various parts of a class, but I mean this:

If I create a button, with a title and color, pressing enter with the cursor right before the color will put it on a new line, with the same indentation as the title.

But if I have a class with a few properties(?), such as

class myClass: this, that, theOther {

}

And I press enter with the cursor right before “that”, the indentation is clearly not supported. It will indent less than the indentation that “this” has, and pressing enter with the cursor right before “theOther” will put no indentation at all.

So in other words, while Xcode makes it easy (which I understand to mean also: supports) multiple in a UIButton declaration, such as:

let myButton = UIButton(title: “blah”,
                                       color: red)

It does not make it easy at all to do so for a class (which I understand to mean also: does not support, or recommend).

I can record a video if further clarification is needed 👍 

Thanks
—

**Comment by jpsim** [comment]
> But if I have a class with a few properties(?), such as `class myClass: this, that, theOther { }`

`this`, `that` and `theOther` are _conformances_. Properties generally refer to member variables of a type. E.g. `class MyClass { var propertyOne: Int }`

If you want to indent `that` on a newline, you can place your cursor at the start of that word, press your keyboard's "Enter" key followed by "Tab".

Here's a gif in case that's not clear:

![tab](https://user-images.githubusercontent.com/474794/50390770-3b667980-06f0-11e9-8120-69ebcbc883e3.gif)

**Comment by igroti** [comment]
Ah ok, so I mean that to indent conformances I have to manually add spaces (or tabs, as you point out), while with most things - like in the above example, UIButtons - the indentation is automatic (I don’t have to press space or tab at all).

Thanks
—

**Comment by igroti** [comment]
Maybe 😅 and I think I’ll report this as a bug, but perhaps it’s intentional - and they don’t think conformances need to break to multiple lines, was my guess.

Thanks
—

**Comment by jpsim** [comment]
Closing then as I don't think SwiftLint should be doing anything differently here.
